pkg/stanza/fileconsumer/internal/fingerprint/fingerprint.go

```diff
@@ -71,13 +87,24 @@ func (f Fingerprint) Equal(other *Fingerprint) bool {
 // a fingerprint. As the file grows, its fingerprint is updated
 // until it reaches a maximum size, as configured on the operator
 func (f Fingerprint) StartsWith(old *Fingerprint) bool {
-	l0 := len(old.FirstBytes)
-	if l0 == 0 {
+	lenOld := old.BytesUsed
+	if lenOld == 0 {
```

Comment:
```suggestion
	if old.BytesUsed == 0 {
```
Maybe it's just me but it seems easier to reason about with the additional allocation.